Allow buffers with no file to be formatted

cichli opened the pull request:
Just found this when I had a scratch buffer with some EDN in that I wanted to format. It's easy to work around using the command for formatting the region, but I don't see any reason for this restriction (seeing as the formatting commands for both code and EDN work on strings alone and not files).

bbatsov commented:
You'll have to rebase this.

bbatsov commented on CHANGELOG.md:
```diff
@@ -42,6 +42,7 @@
 * [#953](https://github.com/clojure-emacs/cider/pull/953) Use `sshx` instead of `ssh` in `cider-select-endpoint`
 * [#956](https://github.com/clojure-emacs/cider/pull/956) Eval full ns form only when needed.
 * Enable annotated completion candidates by default.
+* `cider-format-buffer` and `cider-format-edn-buffer` no longer require the buffer being formatted to be visiting a file.
```
This is an unreleased feature, so this changelog entry is redundant.

cichli commented:
Cool, rebased and updated. Wasn't sure if the changelog entry was needed as some people do use snapshot - though I guess it would have been reported already if it affected anyone :-).

arrdem commented:
:+1:

bbatsov commented:
:+1:
